corehq/apps/es/management/commands/ensure_indices_reindexed.py

```diff
@@ -0,0 +1,165 @@
+            return False
+
+        return False
+
+    @classmethod
+    def _return_missing_index_names(cls):
+        """
+        Returns a tuple of missing primary and secondary index names
+        """
+        missing_primary = []
+        missing_secondary = []
+        for cname in CANONICAL_NAME_ADAPTER_MAP.keys():
+            primary_adapter, secondary_adapter = cls.get_both_adapters_for_cname(cname)
+            if not manager.index_exists(primary_adapter.index_name):
+                missing_primary.append(primary_adapter.index_name)
+            if not manager.index_exists(secondary_adapter.index_name):
+                missing_secondary.append(secondary_adapter.index_name)
+        return missing_primary, missing_secondary
+
+    def is_doc_count_difference_reasonable(self, cname):
+        # We will accept delta of 1% in doc count difference
+        high_frequency_indices = {'case_search', 'cases', 'forms'}
+        primary_adapter, secondary_adapter = self.get_both_adapters_for_cname(cname)
+        delta_perc = self.get_doc_count_delta_percent(cname)
+        if delta_perc == 0:
+            return True
+        elif (
+            primary_adapter.canonical_name in high_frequency_indices
+            and delta_perc <= self.REASONABLE_DIFF_PERCENT
+        ):
+            return True
+        return False
+
+    @classmethod
+    def _primary_index_name(cls, cname):
+        return getattr(const, f'HQ_{cname.upper()}_INDEX_NAME')
+
+    @classmethod
+    def _secondary_index_name(cls, cname):
+        return getattr(const, f'HQ_{cname.upper()}_SECONDARY_INDEX_NAME')
+
+    @classmethod
+    def get_both_adapters_for_cname(cls, cname):
+
+        """
+        returns instance of primary and secondary adapter for the instance
+        """
+        adapter = doc_adapter_from_cname(cname)
+        primary_index_name = cls._primary_index_name(cname)
+        secondary_index_name = cls._secondary_index_name(cname)
+
+        primary_adapter = adapter.__class__(primary_index_name, adapter.type)
+        secondary_adapter = adapter.__class__(secondary_index_name, adapter.type)
+
+        return (primary_adapter, secondary_adapter)
+
+    def get_doc_count_delta_percent(self, cname):
+        """
+        For the given cname the function reutrn difference in count of primary and secondary indices
+        """
+        primary_adapter, secondary_adapter = self.get_both_adapters_for_cname(cname)
+        primary_adapter.delete_tombstones()
+        secondary_adapter.delete_tombstones()
+
+        primary_count = primary_adapter.count({})
+        secondary_count = secondary_adapter.count({})
+
+        print(f"\nDoc Count in old Index {primary_adapter.index_name} - {primary_count}")
+        print(f"\nDoc Count in new Index {secondary_adapter.index_name} - {secondary_count}")
+
+        diff = abs(primary_count - secondary_count)
+        delta = (diff / primary_count) * 100 if primary_count != 0 else 0
+        return delta
```

Comment:
Should this round? Otherwise comparing to zero (0) as in `delta_perc == 0` will always be false if there is any difference at all (even 0.001%). There is another implicit decision here: what is the minimum difference allowed for "low" frequency indexes?
```suggestion
        return round(diff / primary_count * 100, 2) if primary_count != 0 else 0
```

Reply:
I was thinking that low frequency indices should have 0 delta that's why I have not rounded the values. But I am open to using rounding the values which as you have mentioned would change minimum allowed difference to 0.4999%.

Reply:
I think my suggestion would make the minimum allowance < 0.005%

Reply:
Thinking about this more, maybe it is better no to use rounding. I think I'm coming around to your theory that low-volume indexes should have zero difference, and if there is a difference the check should.